FiducialRegistrationWizard/Logic/vtkSlicerFiducialRegistrationWizardLogic.cxx

```diff
@@ -254,32 +254,48 @@ bool vtkSlicerFiducialRegistrationWizardLogic::UpdateCalibration( vtkMRMLNode* n
     return false;
   }
 
-  // Convert the markupsfiducial nodes into vector of itk points
-  vtkNew<vtkPoints> fromPoints;
-  vtkNew<vtkPoints> toPoints;
-  MarkupsFiducialNodeToVTKPoints( fromMarkupsFiducialNode, fromPoints.GetPointer() );
-  MarkupsFiducialNodeToVTKPoints( toMarkupsFiducialNode, toPoints.GetPointer() );
+  // Convert the markupsfiducial nodes into vtk points
+  vtkSmartPointer< vtkPoints > fromPoints = vtkSmartPointer< vtkPoints >::New();
+  MarkupsFiducialNodeToVTKPoints( fromMarkupsFiducialNode, fromPoints );
+  vtkSmartPointer< vtkPoints > toPointsUnordered = vtkSmartPointer< vtkPoints >::New();
+  MarkupsFiducialNodeToVTKPoints( toMarkupsFiducialNode, toPointsUnordered );
+
+  // Determine the order of points and store an "ordered" version of the "To" list
+  vtkSmartPointer< vtkPoints > toPointsOrdered = NULL; // temporary value
+  int inputFormat = fiducialRegistrationWizardNode->GetInputFormat();
+  if ( inputFormat == vtkMRMLFiducialRegistrationWizardNode::INPUT_FORMAT_ORDERED_PAIRS )
+  {
+    toPointsOrdered = toPointsUnordered;
+  }
+  else if ( inputFormat == vtkMRMLFiducialRegistrationWizardNode::INPUT_FORMAT_UNORDERED_PAIRS )
```

Comment:
Often we accidentally record two points or forget to record a point. Can you deal with point sets that have different lengths?